src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/AcspMembershipsController.java

```diff
@@ -50,8 +56,91 @@ public AcspMembershipsController(
+      LOG.infoContext(xRequestId, "User ID for the target user not provided", null);
+      throw new BadRequestRuntimeException(PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN);
+    }
+
+    AcspDataDao acspDataDao;
+    try {
+      acspDataDao = acspDataService.fetchAcspData(acspNumber);
+    } catch (NotFoundRuntimeException exception) {
+      LOG.infoContext(xRequestId, exception.getLocalizedMessage(), null);
+      throw new BadRequestRuntimeException(PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN);
+    }
+
+    this.validateLoggedInUserAuthorisationToAddNewMemberForAcsp(
+        xRequestId, acspNumber, requestBodyPost);
+
+    User user;
+    try {
+      user = usersService.fetchUserDetails(requestBodyPost.getUserId());
+    } catch (NotFoundRuntimeException exception) {
+      LOG.infoContext(xRequestId, exception.getLocalizedMessage(), null);
+      throw new BadRequestRuntimeException(ERROR_CODE_1001.getCode());
+    }
+
+    final var memberships = acspMembersService.fetchAcspMemberships(user, false);
+    if (!memberships.getItems().isEmpty()) {
+      LOG.infoContext(
+          xRequestId,
+          String.format("User with ID %s already has an active ACSP membership", user.getUserId()),
+          null);
+      throw new BadRequestRuntimeException(ERROR_CODE_1002.getCode());
+    }
+
+    final var loggedUser = UserContext.getLoggedUser();
+    final var userRole =
+        AcspMembership.UserRoleEnum.fromValue(requestBodyPost.getUserRole().getValue());
+    final var addedAcspMembership =
+        acspMembersService.addAcspMembership(
+            user,
+            acspDataDao,
+            acspNumber,
+            userRole,
+            Objects.requireNonNull(loggedUser).getUserId());
+
+    return new ResponseEntity<>(addedAcspMembership, HttpStatus.CREATED);
+  }
+
+  private void validateLoggedInUserAuthorisationToAddNewMemberForAcsp(
+      final String xRequestId, final String acspNumber, final RequestBodyPost requestBodyPost) {
+    final var isOauth2 = RequestContextUtil.isOAuth2Request();
+    final var loggedUser = Objects.requireNonNull(UserContext.getLoggedUser());
+    if (isOauth2) {
+      final var loggedUserMemberships = acspMembersService.fetchAcspMemberships(loggedUser, false);
+      final var acspMembership =
+          loggedUserMemberships.getItems().stream()
+              .filter(membership -> membership.getAcspNumber().equals(acspNumber))
+              .findFirst();
+      if (acspMembership.isPresent()) {
+        final var loggedInUserRole = acspMembership.get().getUserRole().getValue();
+        final var newUserRole = requestBodyPost.getUserRole().getValue();
+        final var isStandardUser =
+            loggedInUserRole.equals(AcspMembership.UserRoleEnum.STANDARD.getValue());
+        final var isAdminAddingOwner =
+            loggedInUserRole.equals(AcspMembership.UserRoleEnum.ADMIN.getValue())
+                && newUserRole.equals(AcspMembership.UserRoleEnum.OWNER.getValue());
```

Comment:
```suggestion
        final var newUserRole = requestBodyPost.getUserRole().getValue();
        final var isStandardUser =
            AcspMembership.UserRoleEnum.STANDARD.getValue().equals(loggedInUserRole);
        final var isAdminAddingOwner =
            AcspMembership.UserRoleEnum.ADMIN.getValue().equals(loggedInUserRole)
                && AcspMembership.UserRoleEnum.OWNER.getValue().equals(newUserRole);
```

it's probably better do .equals on the enum since it'll never be null, can avoid a npe - seen it a few times